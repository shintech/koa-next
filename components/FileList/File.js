import React from 'react'
import SVG from 'static/images/react.svg'

const File = ({ file, selectNowPlaying }) => {
  return (
    <a href={`static/files/${file}`} onClick={e => {
      e.preventDefault()
      selectNowPlaying(file)
    }}>
      <div className='cover'><SVG /></div>
      <div className='info'>{file}</div>
    </a>
  )
}

export default File