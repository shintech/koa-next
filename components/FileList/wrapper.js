import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  
  a {
    display: grid;
    border: 1px solid black;
    grid-template-rows: [start row-1] auto [stop end];
    grid-template-columns: [start column-1] auto [column-2] auto [end];
    background-color: slategrey;
    opacity: 0.7;
    
    margin: 0.5ch;
    width: 28ch;
    color: lightgreen;
    
    text-decoration: none;
    border-radius: 1ch;
    
    .cover {
      grid-column: 1;
      grid-row: 1;
      
      svg {
        width: 10ch;
      }
      
    }
    
    .info {
      grid-column: 2;
      grid-row: 1;
      padding-top: 1ch;
    }
    
    :visited {
      color: cyan;
    }
    
    :hover {
      cursor: pointer;
      opacity: 1;
      color: yellow;
    }
  }
`

export default Wrapper
