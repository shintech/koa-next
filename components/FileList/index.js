import PropTypes from 'prop-types'
import File from './File'
import Wrapper from './wrapper'

const FileList = ({ files, selectNowPlaying }) =>
  <Wrapper>
    {files.list.map(file =>  <File selectNowPlaying={selectNowPlaying} key={file} file={file} />)}
  </Wrapper>

FileList.propTypes = {
  files: PropTypes.object.isRequired
}

export default FileList
