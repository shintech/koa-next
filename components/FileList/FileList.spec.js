/* eslint-env jest */

import { shallow } from 'enzyme'
import { compose } from 'redux'
import toJSON from 'enzyme-to-json'
import FileList from 'components/FileList'

const shallowExpect = compose(expect, toJSON, shallow)

// describe('COMPONENT -> Content...', () => {
//   let clicker = shallow(<Content content='Hello World!' />)

//   it(`expect Content p text to equal "Hello World!"...`, () => {
//     expect(clicker.find('p').text()).toBe('Hello World!')
//   })
// })

describe('COMPONENT -> FileList - snapshot...', () => {
  it('expect to render correct properties', () => {
    shallowExpect(<FileList files={{ list: ['file.mp3', 'file.wav'] }} />).toMatchSnapshot()
  })
})
