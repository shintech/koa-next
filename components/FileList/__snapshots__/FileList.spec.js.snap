// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT -> Content - snapshot... expect to render correct properties 1`] = `
<wrapper__Wrapper>
  <File
    file="file.mp3"
    key="file.mp3"
  />
  <File
    file="file.wav"
    key="file.wav"
  />
</wrapper__Wrapper>
`;
